packages/playback-core/src/index.ts

```diff
@@ -41,9 +49,143 @@ export {
+export const updateStreamInfoFromSrc = async (src: string, mediaEl: HTMLMediaElement) => {
+  const playlistLines = await getMediaPlaylistLinesFromMultivariantPlaylistSrc(src);
+
+  const { streamType, targetLiveWindow, liveEdgeStartOffset } = getStreamInfoFromPlaylistLines(playlistLines);
+
+  (muxMediaState.get(mediaEl) ?? {}).liveEdgeStartOffset = liveEdgeStartOffset;
+
+  (muxMediaState.get(mediaEl) ?? {}).targetLiveWindow = targetLiveWindow;
+  mediaEl.dispatchEvent(
+    new CustomEvent('targetlivewindowchange', { composed: true, bubbles: true, detail: targetLiveWindow })
+  );
+
+  (muxMediaState.get(mediaEl) ?? {}).streamType = streamType;
+  mediaEl.dispatchEvent(new CustomEvent('streamtypechange', { composed: true, bubbles: true, detail: streamType }));
+};
+
+export const getStreamInfoFromHlsjsLevelDetails = (levelDetails: any) => {
+  const playlistType: HlsPlaylistTypes = levelDetails.type as HlsPlaylistTypes;
+
+  const streamType = toStreamTypeFromPlaylistType(playlistType);
+  const targetLiveWindow = toTargetLiveWindowFromPlaylistType(playlistType);
+  let liveEdgeStartOffset = undefined;
+  const lowLatency = !!levelDetails.partList?.length;
+  if (streamType === StreamTypes.LIVE) {
+    liveEdgeStartOffset = lowLatency ? levelDetails.partTarget * 2 : levelDetails.targetduration * 3;
+  }
+
+  return {
+    streamType,
+    targetLiveWindow,
+    liveEdgeStartOffset,
+    lowLatency,
+  };
+};
+
+export const updateStreamInfoFromHlsjsLevelDetails = (
+  levelDetails: any,
+  mediaEl: HTMLMediaElement,
+  hls: Pick<Hls, 'config' | 'userConfig' | 'liveSyncPosition'>
+) => {
+  const { streamType, targetLiveWindow, liveEdgeStartOffset, lowLatency } =
+    getStreamInfoFromHlsjsLevelDetails(levelDetails);
+
+  if (streamType === StreamTypes.LIVE) {
+    // Update hls.js config for live/ll-live
+    if (lowLatency) {
+      hls.config.backBufferLength = hls.userConfig.backBufferLength ?? 4;
+      hls.config.maxFragLookUpTolerance = hls.userConfig.maxFragLookUpTolerance ?? 0.001;
+      // For ll-hls, ensure that up switches are weighted the same as down switches to mitigate
+      // cases of getting stuck at lower bitrates.
+      hls.config.abrBandWidthUpFactor = hls.userConfig.abrBandWidthUpFactor ?? hls.config.abrBandWidthFactor;
+    } else {
+      hls.config.backBufferLength = hls.userConfig.backBufferLength ?? 8;
+    }
+
+    // Proxy `seekable.end()` to constrain based on rules in
+    // https://github.com/video-dev/media-ui-extensions/blob/main/proposals/0007-live-edge.md#property-constraint-on-htmlmediaelementseekableend-to-model-seekable-live-edge
+    const seekable: TimeRanges = Object.freeze({
+      get length() {
+        return mediaEl.seekable.length;
+      },
+      start(index: number) {
+        return mediaEl.seekable.start(index);
+      },
+      end(index: number) {
+        if (index > this.length) return mediaEl.seekable.end(index);
+        return hls.liveSyncPosition ?? mediaEl.seekable.end(index);
```

Comment:
is this correct? all indexes will return liveSyncPosition if it exists

Reply:
Yes that's by design per https://github.com/video-dev/media-ui-extensions/blob/main/proposals/0007-live-edge.md#property-constraint-on-htmlmediaelementseekableend-to-model-seekable-live-edge